File: src/test/java/dev/jiaming/ai_interview/common/RuntimeModePropertiesTests.java
package dev.jiaming.ai_interview.common;

import static org.assertj.core.api.Assertions.assertThat;
import static org.assertj.core.api.Assertions.assertThatThrownBy;

import org.junit.jupiter.api.Test;

class RuntimeModePropertiesTests {

	@Test
	void exposesApiAndWorkerCapabilities() {
		assertThat(new RuntimeModeProperties("all").apiEnabled()).isTrue();
		assertThat(new RuntimeModeProperties("all").workerEnabled()).isTrue();
		assertThat(new RuntimeModeProperties("api").workerEnabled()).isFalse();
		assertThat(new RuntimeModeProperties("worker").apiEnabled()).isFalse();
	}

	@Test
	void rejectsUnknownMode() {
		assertThatThrownBy(() -> new RuntimeModeProperties("batch"))
			.isInstanceOf(IllegalArgumentException.class)
			.hasMessageContaining("all, api, or worker");
	}
}

File: src/test/kotlin/dev/jiaming/ai_interview/common/RuntimeModePropertiesTests.kt
package dev.jiaming.ai_interview.common

import org.assertj.core.api.Assertions.assertThat
import org.assertj.core.api.Assertions.assertThatThrownBy
import org.junit.jupiter.api.Test

class RuntimeModePropertiesTests {
	@Test
	fun exposesApiAndWorkerCapabilities() {
		assertThat(RuntimeModeProperties("all").apiEnabled()).isTrue()
		assertThat(RuntimeModeProperties("all").workerEnabled()).isTrue()
		assertThat(RuntimeModeProperties("api").workerEnabled()).isFalse()
		assertThat(RuntimeModeProperties("worker").apiEnabled()).isFalse()
	}

	@Test
	fun rejectsUnknownMode() {
		assertThatThrownBy { RuntimeModeProperties("batch") }
			.isInstanceOf(IllegalArgumentException::class.java)
			.hasMessageContaining("all, api, or worker")
	}
}
